### ISSUE TITLE
Add "Field Groups" in the UI to allow expand/collapse of multiple fields organized together

### ISSUE THREAD

**Issue by skyfox675** [twentyhq/twenty#5241]
## Scope & Context

As a user, having many fields is quite cumbersome and hard to find what I am looking for. I would like to have a way to parent/child multiple fields into groups per-say that allows expanding/collapsing of them to make working with large sets of fields more usable. 

As an example, this is how I am doing it today in another CRM:
![image](https://github.com/twentyhq/twenty/assets/4504420/39cbded9-09e1-46a8-9790-d0560e4a558e)

**Comment by FelixMalfait** [comment]
Yes I'd love this too!!! This is planned as part of a big refactoring of the layout engine. This will only come towards the end of year unfortunately
